backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/pageindicator/BpkPageIndicator.kt

```diff
@@ -0,0 +1,151 @@
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.lazy.LazyRow
+import androidx.compose.foundation.lazy.itemsIndexed
+import androidx.compose.foundation.lazy.rememberLazyListState
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.ExperimentalComposeUiApi
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.semantics.invisibleToUser
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.unit.Dp
+import kotlinx.coroutines.launch
+import net.skyscanner.backpack.compose.theme.BpkTheme
+import net.skyscanner.backpack.compose.tokens.BpkSpacing
+import java.lang.Integer.min
+
+enum class BpkPageIndicatorStyle {
+  Default,
+  OverImage,
+}
+
+@Composable
+fun BpkPageIndicator(
+  currentIndex: Int,
+  totalIndicators: Int,
+  modifier: Modifier = Modifier,
+  style: BpkPageIndicatorStyle = BpkPageIndicatorStyle.Default,
+) {
+
+  if (totalIndicators <= 1) throw IllegalArgumentException("totalIndicators must be greater than 1")
+  if (currentIndex !in 0 until totalIndicators) throw IllegalArgumentException("currentIndex must be between 0 and $totalIndicators")
+  val indicatorSize = BpkSpacing.Md
+  val indicatorCount = min(totalIndicators, DISPLAY_DOTS_MAX)
+
+  val coroutineScope = rememberCoroutineScope()
+  val state = rememberLazyListState()
+
+  val offsetCount = if (totalIndicators > DISPLAY_DOTS_MAX) {
+    when (currentIndex) {
+      0 -> 0
+      1 -> 2
+      totalIndicators - 2 -> 6
+      totalIndicators - 1 -> 8
+      else -> 4
+    }
+  } else {
+    0
+  }
+  val offsetPx = with(LocalDensity.current) { (indicatorSize * offsetCount).toPx().toInt() }
+  LaunchedEffect(currentIndex) {
+    coroutineScope.launch {
+      state.animateScrollToItem(currentIndex, -offsetPx)
+    }
+  }
+
+  LazyRow(
+    modifier = modifier.pageIndicatorModifier(indicatorCount, indicatorSize),
+    state = state,
+    horizontalArrangement = Arrangement.Center,
+    verticalAlignment = Alignment.CenterVertically,
+    userScrollEnabled = false,
+  ) {
+    itemsIndexed(
+      items = (0 until totalIndicators - 1).toList(),
+      key = { index, _ -> index.hashCode() },
+    ) { index, _ ->
+      PageIndicatorDot(
+        indicatorSize = indicatorSize,
+        isSelected = index == currentIndex,
+        style = style,
+      )
+      Spacer(modifier = Modifier.width(indicatorSize))
+    }
+    item {
```

Comment:
I was putting a spacer in between items because lists' horizontalArrangement = spacedBy was causing issues with RTL layouts. But now I removed it and added padding to the item itself